.changesets/fix_tninesling_cost_result_filtering.md

```diff
@@ -0,0 +1,41 @@
+### Fix cost result filtering for custom metrics ([PR #5838](https://github.com/apollographql/router/pull/5838))
+
+Fix filtering for custom metrics that use demand control cost information in their conditions. This allows a telemetry config such as:
+
+```yaml
+telemetry:
+  instrumentation:
+    instruments:
+      supergraph:
+        cost.rejected.operations:
+          type: histogram
+          value:
+            cost: estimated
+          description: "Estimated cost per rejected operation."
+          unit: delta
+          condition:
+            eq:
+              - cost: result
+              - "COST_ESTIMATED_TOO_EXPENSIVE"
+```
+
+Additionally, this fixes an issue with attribute comparisons which would silently fail to compare integers to float values. Now, users can write integer values in conditions that compare against selectors that select floats:
```

Comment:
```suggestion
This also fixes an issue where attribute comparisons would fail silently when comparing integers to float values. Users can now write integer values in conditions that compare against selectors that select floats:
```